### PR TITLE
AST-3365 - Widget section need to provide font color and font size option

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

Added Widget design options as a default.

### Screenshots

- https://d.pr/v/5zn66E

### How has this been tested?

1. Check the footer widget design option - It should be visible and working for respective blocks like the navigation menu, normal paragraph, heading

2. Check the same above case with the Addon module activated - 
A. When the Addon module is activated that time more typography options should be visible like font size font selection, and decoration option.
B. When Addon is not activated then for typography just the font size option should be visible.

3. Check the above same for the header by adding widgets.

4. After enabling the Addon module sticky header widget design option should be visible.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/lib/astra-notices/class-astra-notices.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function dismiss_notice() {
 			}
 
 			if ( false === wp_verify_nonce( $nonce, 'astra-notices' ) ) {
-				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.', 'astra' ) );
+				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.' ) );
```
[the remark this answers is not in the document]
Comment (yssambare12) [reply inc/lib/astra-notices/class-astra-notices.php]: Updated!

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php [hunk 2]
```diff
@@ -103,6 +101,27 @@ public static function astra_widget_dynamic_css( $builder_type = 'header' ) {
...
+					$selector => array(
+						// Margin CSS.
+						'margin-top'    => astra_responsive_spacing( $margin, 'top', 'mobile' ),
+						'margin-bottom' => astra_responsive_spacing( $margin, 'bottom', 'mobile' ),
+						'margin-left'   => astra_responsive_spacing( $margin, 'left', 'mobile' ),
+						'margin-right'  => astra_responsive_spacing( $margin, 'right', 'mobile' ),
+					),
+				);
+
+
```
[the remark this answers is not in the document]
Comment (yssambare12) [reply inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:124]: @imnavanath Right! This is unnecessary CSS, as we already adding responsive device margin CSS. Added else part mistakenly.
Thanks for pointing out.

[the remark this answers is not in the document]
Comment (yssambare12) [reply inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php]: Updated.

**Comment by imnavanath** [comment]
@yssambare12 changelog is missing.

**Comment by yssambare12** [comment]
> @yssambare12 changelog is missing.

Added!

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php [hunk 3]
```diff
@@ -94,6 +98,31 @@ public static function astra_widget_dynamic_css( $builder_type = 'header' ) {
...
+						'font-size' => astra_responsive_font( $heading_font_size, 'desktop' ),
+					),
+					$selector . ' .uagb-timeline__heading'          => array(
+						'color'     => $heading_color_desktop,
+						'font-size' => astra_responsive_font( $heading_font_size, 'desktop' ),
+					),
+					$selector . ' .uagb-notice-title'          => array(
+						'color'     => $heading_color_desktop,
+						'font-size' => astra_responsive_font( $heading_font_size, 'desktop' ),
+					),
```
Comment (imnavanath) [inline inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:125]: Why so much lines of code? What if any other heading element is added in widget area apart from this??

File: inc/builder/type/base/assets/js/customizer-preview.js [hunk 4]
```diff
@@ -600,18 +600,20 @@ function astra_builder_widget_css( builder_type = 'header' ) {
...
 
-		// Widget Title Color.
+
+		// Widget Heading Color.
 		astra_color_responsive_css(
 			builder_type + '-widget-' + index + '-title-color',
 			'astra-settings[' + builder_type + '-widget-' + index + '-title-color]',
 			'color',
-			selector + ' .widget-title'
+			selector + ' .widget-title, ' + selector + ' .wp-block-heading, ' + selector + ' .uagb-ifb-title, ' + selector + ' .uagb-heading-text, ' + selector + ' .uagb-cta__title, ' + selector + ' .uagb-timeline__heading, ' + selector + ' .uagb-notice-title '
```
Comment (imnavanath) [inline inc/builder/type/base/assets/js/customizer-preview.js:609]: What about for other headings if added in widget area?

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php [hunk 5]
```diff
@@ -114,6 +143,11 @@ public static function astra_widget_dynamic_css( $builder_type = 'header' ) {
 						// Typography.
 						'font-size' => astra_responsive_font( $title_font_size, 'tablet' ),
 					),
+					$selector . ' .wp-block-heading'          => array(
```
Comment (imnavanath) [inline inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:146]: what about other selectors you used above? why not add their responsiveness?

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php [hunk 6]
```diff
@@ -140,6 +174,11 @@ public static function astra_widget_dynamic_css( $builder_type = 'header' ) {
 						// Typography.
 						'font-size' => astra_responsive_font( $title_font_size, 'mobile' ),
 					),
+					$selector . ' .wp-block-heading'          => array(
```
[the remark this answers is not in the document]
Comment (yssambare12) [reply inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:177]: implemented here as per suggestion - https://github.com/brainstormforce/astra/pull/5389/files

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php (hunk 5 above)
Comment (yssambare12) [reply inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:146]: implemented here as per suggestion - https://github.com/brainstormforce/astra/pull/5389/files

File: inc/builder/type/base/assets/js/customizer-preview.js (hunk 4 above)
Comment (yssambare12) [reply inc/builder/type/base/assets/js/customizer-preview.js:609]: implemented here as per suggestion - https://github.com/brainstormforce/astra/pull/5389/files

File: inc/builder/type/base/assets/js/customizer-preview.js [hunk 7]
```diff
@@ -600,18 +600,20 @@ function astra_builder_widget_css( builder_type = 'header' ) {
...
 		);
 
+
 		// Widget Title Typography.
 		astra_responsive_font_size(
 			'astra-settings[' + builder_type + '-widget-' + index + '-font-size]',
-			selector + ' .widget-title'
+			selector + ' .widget-title, ' + selector + ' .wp-block-heading'
```
[the remark this answers is not in the document]
Comment (yssambare12) [reply inc/builder/type/base/assets/js/customizer-preview.js:616]: implemented here as per suggestion - https://github.com/brainstormforce/astra/pull/5389/files

File: inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php (hunk 3 above)
Comment (yssambare12) [reply inc/builder/type/base/dynamic-css/widget/class-astra-widget-component-dynamic-css.php:125]: implemented here as per suggestion - https://github.com/brainstormforce/astra/pull/5389/files

**Comment by yssambare12** [comment]
Closing this PR due to multiple critical conflicts & continuing here- https://github.com/brainstormforce/astra/pull/5389/files

**Pull request closed without merging**